### ISSUE TITLE
Show and Tell: A Neural Image Caption Generator

### ISSUE THREAD

**Issue by reyllama** [reyllama/papers#5]
## Authors

Oriol Vinyals, Alexander Toshev, Samy Bengio, Dumitru Erhan

## Publication Year

2015

## Tags

- Natural Language Processing
<br>

- Computer Vision
<br>

- Generative Model

## Summary

- Image Captioning is harder than traditional image recognition in several ways:

> 1) It must capture not only the objects but also the relationship, attributes and actions of them. 

> 2) The output has to make sense as  **natural language.**

- The idea was inspired from **Language Translation Model using Recurrent Neural Network.**

- Instead of using <RNN encoder - RNN decoder> pair, they used **<CNN encoder - RNN decoder>** pair to take image as input and yield natural language sentences.

- Just like state-of-the-art statistical machine translation, they employ **'end-to-end'** fashion learning algorithm, under which it tries directly to maximize the probability of correct sentence output given image input. 

<br>

![image](https://user-images.githubusercontent.com/46308407/77838829-6e7ca000-71b2-11ea-9ab4-485887c4bee5.png)

- S is unbounded in terms of length, so chain rule is used to model the joint probablity over S_0, S_1, S_2, ... , S_N.

- For RNN model, **LSTM**, which handles the vanishing and exploding gradient problem most effectively. is used.

![image](https://user-images.githubusercontent.com/46308407/77853635-fea4fe80-721f-11ea-8998-81fd229cf84e.png)

- The core of LSTM is the memory cell **c** that is governed by 3 **"gates"**, each representing forget, input and output. This architecture grants robustness against vanishing/exploding gradient problem.

![image](https://user-images.githubusercontent.com/46308407/77853623-e3d28a00-721f-11ea-9167-0ca66031b22c.png)

- Output words are chosen from a given dictionary, and therefore represented as fixed length vector its dimension equal to that of the dictionary. 

- Both the image(CNN) and the words(Word Embedding) are mapped to the same space.

- The loss function is given as sum of negative log likelihood of correct word at each step:

![image](https://user-images.githubusercontent.com/46308407/77852708-14fc8b80-721b-11ea-858a-7d81cc2cf24a.png)

- Parameters contributing to the minimization of the above loss function comes from **LSTM**, top layer of **image embedding CNN** and **word embedding**.

- For inference, **Beam Search** of beam size 20 was used.

>- Beam Search: iteratively take *k* best sentences up to time *t* as candidates to generate sentence of size *t+1*.

- Evaluating descriptions is rather an ambiguous task. They relied on *(1) human rating*: 2 people giving scores ranging 1-4, and *(2) BLEU score*.

- Overfitting occurred due to limited dataset, so they took mitigation measures:

> (1) Initializing the weights of CNN components to a pretrained model (ex: ImageNet).

> (2) Dropout and Ensembling.

- **Stochastic Gradient Descent**, **Fixed Learning Rate** and **Random Parameter Initialization** (except for CNN, left unchanged from its pretrained reference model)

- Transfer learning dragged down the BLEU score to a certain extent, but they report the descriptions are still rather *reasonable*.

- N-best generated output list shows this generative model does attain the virtue of **diversity**. They reflect diverse angles of a given image but still remain relatively reasonable. Letting N=15, they encountered quite a lot (1 out of 2) of novel descriptions yet the BLEU score remained about the same.

![image](https://user-images.githubusercontent.com/46308407/77853279-50e52000-721e-11ea-9c5c-d9ff1b8ac887.png)

- They reported some success in word embedding: it seemed to have captured important representations in words. 

![image](https://user-images.githubusercontent.com/46308407/77853363-cf41c200-721e-11ea-96fa-e77e737de044.png)

- Some output descriptions of each rating class.

![image](https://user-images.githubusercontent.com/46308407/77853643-12e8fb80-7220-11ea-9af0-70c76845c7e2.png)

## Personal Comments

- I was personally hoping for the emergence of a DNN model that generates image that suits an instruction in human language, and this article provided path from the opposite direction. I guess my dreams are not in distant future.

- Perhaps because this paper was published 5 years ago, everything seemed quite intuitive and clear. Using CNN and LSTM respectively for image encoding and sentence generating, putting them together like a variational auto encoder. (Maybe there are some differences but for now I'm not sure)

- Looking at the generated captions they provided, all sentences were written in extremely plain language, and though some of them accurately addressed the image, I personally thought the accuracy wasn't that satisfactory for this paper itself. Since 5 years have passed now, I might find some astonishing advancements somewhere.

- I was not sure about the meaning of **word embedding**.

- I learned briefly about **LSTM** on MIT Intro to Deep Learning lecture as well, but I don't think I fully understood its structures.
